shed/event_streams.py

```diff
@@ -381,7 +385,11 @@ def start(self, docs, clear=True):
             self._clear()
         self.run_start_uid = str(uuid.uuid4())
         self.parent_uids = [doc['uid'] for doc in docs if doc]
-        new_start_doc = dict(uid=self.run_start_uid,
+        new_start_doc = dict()
+        if self.keep_start:
+            for doc in docs:
```

Comment:
Note that this has right to left resolution, rather than left to right.

Reply:
thanks, noted. it should conform with how `doc_contents` resolution order to be consisten